src/cart/src/cart/crt_hg_proc.c

```diff
@@ -44,18 +44,40 @@
 #include "crt_internal.h"
 
 #define CRT_PROC_NULL (NULL)
+#define CRT_PROC_TYPE_FUNC(type) \
+	int crt_proc_##type(crt_proc_t proc, type *data) \
+	{ \
+		crt_proc_op_t	 proc_op; \
+		type		*buf; \
+		int		 rc = 0; \
+		rc = crt_proc_get_op(proc, &proc_op); \
+		if (unlikely(rc)) \
+			return -DER_HG; \
+		buf = hg_proc_save_ptr(proc, sizeof(*buf)); \
+		switch (proc_op) { \
+		case CRT_PROC_ENCODE: \
+			*buf = *data; \
+			break; \
+		case CRT_PROC_DECODE: \
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment